src/sql/workbench/services/connection/browser/connectionManagementService.ts

```diff
@@ -1231,6 +1231,10 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 						if (connectionMngInfo.serverInfo) {
 							connection.options.isCloud = connectionMngInfo.serverInfo.isCloud;
 						}
+						else if (connectionMngInfo.ownerUri.indexOf('|user:|') !== -1) {
```

Comment:
if you want to check whether username is provided, why not check the connectioninfo directly rather than have a dependency on a string representation of the connection which might to subject to changes.